src/buffer.rs

```diff
@@ -91,8 +92,8 @@ impl<'a> Buffer<'a> {
     }
 
     /// The length of the buffer in bytes.
-    pub fn length(&self) -> usize {
-        self.json.byte_length as usize
+    pub fn length(&self) -> u64 {
```

Comment:
This is just going to lead to casts every time it is used. I think `usize` was the right choice. Buffers exceeding `u32::MAX` length become problematic on 32bit systems. I don't think you can even address a slice in Rust beyond `u32::MAX` on 32bit systems. 

I agree that internally, the length should be stored as `u64` but I think the offsets and lengths should be reported as `usize`.

Reply:
Loading the gltf file should fail on 32bit systems if there are buffers with lengths larger than `u32::MAX`.

Reply:
Hmm, I figured it's worth preserving the actual information losslessly as long as possible so users of the interface can decide how they want to deal with it. Do you want to truncate the information here by casting to usize, which may lead to unexpected behaviour, or should this function be fallible?

Reply:
@aloucks Do you have an update on how you would prefer this to be handled? It sounds like you'd like to fail when loading a glTF file that has any buffer that is too large and then just hard cast to usize here later? The use case of working with larger files on 32 bit systems is not worth supporting?

Reply:
> Do you want to truncate the information here by casting to usize, which may lead to unexpected behaviour, or should this function be fallible?

No, I was saying that it should fail to load if the buffer length exceeds `usize::MAX` and therefor truncation here could never happen. 

If you want to be able to load the json metadata on 32 bit systems that have buffers that exceed `u32::MAX` length, then I guess reporting it as `u64` is fine. My original thought process was that this is just going to shove the cast onto the user, forcing them them to deal with the error condition, when that condition can't even happen on a 64bit system that could load the file (with buffers).

Reply:
@aloucks I'm sorry to nag you yet again, but would you like me to change this or anything else? I need to handle large files in a project, and while I can of course just use this PR for the crate version, I would prefer to use an official release. It's not very urgent, but it would be nice if we could either merge or shelve this PR this year so I don't have to think about it anymore.

Reply:
I'm tempted to settle on `usize` as 32 bit systems are the exception rather than the norm nowadays. The [WASM](https://webassembly.org/) runtime however is a popular target for this library and [64 bit addressing](https://webassembly.org/roadmap/#linear-memory-bigger-than-4-gib) is unstable. For this reason, I think it's important the user can handle this scenario gracefully somehow.

For the majority of users running on 64 bit systems, the cast from `u32` to `usize` is already a bit annoying, so I suggest few things:
1. Create a new type `struct Address(u64)` (the name is debatable) in gltf-json.
2. Implement `Validate` for `Address` and report an error on 32 bit systems if it exceeds the `usize` range.
3. In the main/wrapper crate, change the return type for existing `u32` offsets/sizes to `usize`, using `as` to cast internally.

This should cover all of the points mentioned previously. 32 bit systems will still be able to process large glTF (and the wrapper via. [`from_json_without_validation`](https://docs.rs/gltf/latest/gltf/struct.Document.html#method.from_json_without_validation)) or report to the user the out of range size otherwise.

@ShaddyDC, would you like to implement this yourself? I can chip in and push to this branch if you like otherwise.

Reply:
I can take a shot at an implementation, but if you've got the time to do it, I'd appreciate if you could take care of it. I'm not yet familiar with the validation implementation.

Reply:
Leave it with me :smile:

Reply:
I've opened a PR on your fork: https://github.com/ShaddyDC/gltf/pull/1

Reply:
Lgtm, merged!